doc/manipulating_images/input_output.rst

```diff
@@ -0,0 +1,258 @@
+
+    The two different environment variables (NILEARN_SHARED_DATA and
+    NILEARN_DATA) are provided for multi-user systems, to distinguish a
+    global dataset repository that may be read-only at the user-level.
+    Note that you can copy that folder to another user's computers to
+    avoid the initial dataset download on the first fetching call.
+
+
+
+|
+
+Understanding neuroimaging data
+===============================
+
+Nifti and Analyze data
+-----------------------
+
+For volumetric data, nilearn works with data stored as in the Nifti
+structure (via the nibabel_ package).
+
+The `NifTi <http://nifti.nimh.nih.gov/>`_ data structure (also used in
+Analyze files) is the standard way of sharing data in neuroimaging
+research. Three main components are:
+
+:data:
+    raw scans in form of a numpy array: ``data = img.get_data()``
+:affine:
+    returns the transformation matrix that maps
+    from voxel indices of the numpy array to actual real-world
+    locations of the brain:
+    ``affine = img.get_affine()``
+:header:
+    low-level informations about the data (slice duration, etc.):
+    ``header = img.get_header()``
+
+If you need to load the data without using nilearn, read the nibabel_
+documentation.
+
+
+.. topic:: **Dataset formatting: data shape**
+
+    It is important to appreciate two main representations for
+    storing and accessing more than one Nifti images, that is sets
+    of MRI scans:
+
+    - a big 4D matrix representing (3D MRI + 1D for time), stored in a single
+      Nifti file.
+      `FSL <http://www.fmrib.ox.ac.uk/fsl/>`_ users tend to
+      prefer this format.
+    - several 3D matrices representing each time point (single 3D volume) of the
+      session, stored in set of 3D Nifti or analyse files.
+      `SPM <http://www.fil.ion.ucl.ac.uk/spm/>`_ users tend
+      to prefer this format.
+
+.. _niimg:
+
+Niimg-like objects
+-------------------
+
+Nilearn functions take as input argument what we call "Niimg-like
+objects":
+
+**Niimg:** A Niimg-like object can be one of the following:
+
+  * A string with a file path to a Nifti or Analyse image
+  * Any object exposing ``get_data()`` and ``get_affine()`` methods, typically
+    a ``Nifti1Image`` from nibabel_.
+
+**Niimg-4D:** Similarly, some functions require 4D Nifti-like
+data, which we call Niimgs or Niimg-4D. Accepted input arguments are:
+
+  * A path to a 4D Nifti image
+  * List of paths to 3D Nifti images
+  * 4D Nifti-like object
+  * List of 3D Nifti-like objects
+
+.. topic:: **Image affines**
+
+   If you provide a sequence of Nifti images, all of them must have the same
+   affine.
+
+Text files: phenotype or behavior
+----------------------------------
+
+Phenotypic or behavioral data are often provided as text or CSV
+(Comma Separated Values) file. They
+can be loaded with `numpy.genfromtxt` but you may have to specify some options
+(typically `skip_header` ignores column titles if needed).
```

Comment:
+1 it is the only one to handle string fields correctly. In that case, I would not mention `gen_fom_txt` at all.